flow/aggregates.py

```diff
@@ -278,35 +279,35 @@ def __init__(self, aggregator, project):
         self._aggregator = aggregator
 
         # We need to register the aggregates for this instance using the
-        # project provided.
-        self._aggregates = []
-        self._aggregate_ids = {}
+        # project provided. After registering, we store the aggregates
+        # mapped with the ids using the `get_aggregate_id` method.
+        self._aggregate_per_id = OrderedDict()
         self._register_aggregates(project)
 
     def __iter__(self):
-        yield from self._aggregates
+        yield from self._aggregate_per_id.values()
```

Comment:
@bdice and @vyasr are suggesting that `AggregateStore` objects behave like mappings, so this should return ids.

We need to change the way `DefaultAggregateStore` works as well to mimic this behavior. To iterate over values we need to implement a values function. These classes should be using `MutableMapping` as their parent class given our current decision.

Reply:
This has been resolved.